pkg/openai/client.go

```diff
@@ -383,6 +394,16 @@ func (c *Client) Call(ctx context.Context, messageRequest types.CompletionReques
 		return nil, err
 	} else if !ok {
 		response, err = c.call(ctx, request, id, status)
+
+		// If we got back a context length exceeded error, keep retrying and shrinking the message history until we pass.
+		var apiError *openai.APIError
+		if err != nil && errors.As(err, &apiError) && apiError.Code == "context_length_exceeded" && messageRequest.Chat {
```

Comment:
nit: `errors.As` takes care of the `err != nil` check
```suggestion
		if errors.As(err, &apiError) && apiError.Code == "context_length_exceeded" && messageRequest.Chat {
```

Reply:
Fixed